### PR TITLE
[Templating] Remove the component from the core

### PR DESCRIPTION
**Pull request by dosten**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | part of #15028 |
| License | MIT |
| Doc PR | - |

Hi there!

This PR is part of the extraction of the Templating component, to keep BC i've added `symfony/templating` as a dependency in the root `composer.json`.

### CODE REVIEW

**Comment by dosten** [comment]
Rebased with the latest changes in 2.8.

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
         "php": ">=5.3.9",
         "doctrine/common": "~2.3",
         "twig/twig": "~1.18",
-        "psr/log": "~1.0"
+        "psr/log": "~1.0",
+        "symfony/templating": "~2.8"
```
Comment (fabpot) [inline composer.json]: I would instead add it in FrameworkBundle composer.json file. What do you think?

Comment (dosten) [inline composer.json]: The dependency [already exists in the FrameworkBundle](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/composer.json#L31), which is ok. We need to add this dependency here to keep BC, so, if someone requires the `symfony/symfony:~2.8` package can use the Templating component without any extra change in the `composer.json` file.

**Comment by dosten** [comment]
@fabpot If you agree with my previous comment I think that we can merge this one.

**Comment by jakzal** [comment]
@dosten rebase is needed here again :)

**Comment by dosten** [comment]
@jakzal done!

**Comment by dosten** [comment]
The feature freeze is soon. We should make a decision.

File: composer.json (hunk 1 above)
Comment (Taluu) [inline composer.json]: I'm not quite sure it should be removed in 2.8 rather than 3.0 (where this line will be useless basically)...

Comment (dosten) [inline composer.json]: @Taluu the idea is to remove the componentes un 2.8, add it as a dependency and provide a templating bundle like the security acl

Comment (Taluu) [inline composer.json]: How about putting it in the suggest section then ? And as @fabpot said, I think the dependency in the framework bundle is enough (having to add one line to the composer.json file is fine I think, with a proper upgrade message ofc)

Comment (dosten) [inline composer.json]: We can't add it to the suggest section to keep BC. In 3.0 we can.
The dep in the  framework bundle is not enough, only works if someone install the symfony/framework-bundle package, but if someone uses the full framework requiring symfony/symfony the dep will not be installed.

**Comment by fabpot** [comment]
Before doing this, it should work well, which is (unfortunately) not the case yet. See #15970 for a first PR that improve the current situation.

**Comment by xabbuh** [comment]
Shouldn't we also trigger a deprecation when developers use the `Bundle::Controller::template` syntax?

**Comment by fabpot** [comment]
@xabbuh Indeed, that would be a nice step towards the end goal.

**Comment by fabpot** [comment]
Closing this PR as we won't be ready to merge this for 3.0, there are still to much feature that need to be deprecated first. We can revisit this later on of course.

**Comment by javiereguiluz** [comment]
A quick question: could this deprecation be done in 3.1, 3.2, etc. or should we wait for 4.0?

**Comment by stof** [comment]
@javiereguiluz deprecations can happen in any minor version. Removal happens in next major.

Btw, deprecating in the new major version looks like the forbidden case: the major version is here to remove deprecated layers, not to deprecate new things itself (things should be deprecated in the last minor version before it, like 2.8 vs 3.0)

**Comment by fabpot** [comment]
We can definitely deprecate this in 3.1.

**Pull request closed without merging**